### ISSUE TITLE
[CHORE] Upgrade to latest EKS Major Version

### ISSUE THREAD

**Issue by ben851** [cds-snc/notification-planning-core#442]
# What

EKS 1.31 has been released, we should verify there are no breaking changes and begin the upgrade path.

# Why

It is important to stay up to date with EKS to avoid technical debt, and increase security.

# Acceptance Criteria

- [x] Changelog for EKS 1.31 has been analyzed and determined that there are no breaking changes (or correct any changes)
- [x] Dev upgraded to EKS 1.31
- [x] Staging upgraded to EKS 1.31
- [x] Performance test on staging to verify no impact
- [x] Production upgraded to EKS 1.31
# Resources

# Triggers / Schedule

Ad-hoc based on EKS release schedule

https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions-standard.html

**Comment by ben851** [comment]
No significant deprecations for 1.31

**Comment by ben851** [comment]
Dev upgraded to k8s 1.31 and looks good. 

Prod has a pending 1.30 node version update, so I'm going to put this to blocked until that is completed.

**Comment by sastels** [comment]
Steve will do a PR to bring prod up to latest 1.30

**Comment by sastels** [comment]
Will upgrade staging today, leave over the weekend, and then upgrade prod next week.

**Comment by sastels** [comment]
staging went fine, staging perf test fine.

**Comment by ben851** [comment]
PR needs to get approved so we can release today
https://github.com/cds-snc/notification-terraform/pull/1623

**Comment by sastels** [comment]
Steve will check version in prod as QA

**Comment by sastels** [comment]
```
➜  Git kubectl get nodes

NAME                                            STATUS   ROLES    AGE     VERSION
ip-10-0-123-225.ca-central-1.compute.internal   Ready    <none>   45h     v1.31.0-eks-a737599
ip-10-0-36-20.ca-central-1.compute.internal     Ready    <none>   6h20m   v1.31.0-eks-a737599
ip-10-0-41-117.ca-central-1.compute.internal    Ready    <none>   45h     v1.31.0-eks-a737599
ip-10-0-49-227.ca-central-1.compute.internal    Ready    <none>   3h12m   v1.31.0-eks-a737599
ip-10-0-50-66.ca-central-1.compute.internal     Ready    <none>   3h4m    v1.31.0-eks-a737599
ip-10-0-58-233.ca-central-1.compute.internal    Ready    <none>   45h     v1.31.0-eks-a737599
ip-10-0-62-33.ca-central-1.compute.internal     Ready    <none>   6h16m   v1.31.0-eks-a737599
ip-10-0-66-0.ca-central-1.compute.internal      Ready    <none>   45h     v1.31.0-eks-a737599
ip-10-0-91-197.ca-central-1.compute.internal    Ready    <none>   45h     v1.31.0-eks-a737599
```

**Comment by ben851** [comment]
Upgrade to EKS 1.32 
 
Requires an upgrade to Karpenter 
https://github.com/cds-snc/notification-manifests/pull/3707

**Comment by P0NDER0SA** [comment]
Deployed to dev -- will verify it there before we move forward with Staging etc

**Comment by ben851** [comment]
Been sitting in dev for 3 weeks with no issues. Looking to move to staging.

**Comment by ben851** [comment]
I moved the Karpenter work to its own card #562

Karpenter is upgraded in dev and staging, and needs to be upgraded in production.

Dev is running on 1.32

Once #562 is done, we can upgrade to 1.32 in staging and production.

**Comment by P0NDER0SA** [comment]
We are going to push Karpenter to prod and then look at doing this upgrade on Staging so that we have less things to debug if something goes wrong before pushing the other upgrades.

**Comment by ben851** [comment]
Upgrading staging to 1.32
https://github.com/cds-snc/notification-terraform/pull/1868

**Comment by ben851** [comment]
Staging upgraded to 1.32. Will let it run w/ some perf tests this aft

**Comment by ben851** [comment]
Staging looks good,
Prod PR
https://github.com/cds-snc/notification-terraform/pull/1877

**Comment by ben851** [comment]
To be released today

**Comment by ben851** [comment]
Dev looks good. 
Upgrading to 1.33 in staging
https://github.com/cds-snc/notification-terraform/pull/1940

**Comment by ben851** [comment]
Will aim to do this today in prod (1.33)

**Comment by P0NDER0SA** [comment]
QA'ed!  looks good :thumbsup:
